src/core/packages/chrome/browser-internal/src/ui/project/sidenav_v2/navigation/navigation.tsx

```diff
@@ -0,0 +1,405 @@
+              return {
+                id: child.id,
+                label: warnIfMissing(child, 'title', 'Missing Title 😭'),
+                href: warnIfMissing(child, 'href', 'Missing Href 😭'),
+                external: child.isExternalLink,
+                iconType: child.icon,
+                'data-test-subj': getTestSubj(child),
+              };
+            }),
+          },
+        ];
+      } else {
+        // Otherwise, we need to create sections for each child
+        secondarySections = filterEmpty(
+          navNode.children.map((child) => {
+            if (child.sideNavStatus === 'hidden') return null;
+            if (!child.children?.length) return null;
+
+            warnUnsupportedNavNodeOptions(child);
+
+            const secondaryItems: SecondaryMenuItem[] =
+              child.children
+                .filter((subChild) => subChild.sideNavStatus !== 'hidden')
+                .map((subChild) => {
+                  warnUnsupportedNavNodeOptions(subChild);
+                  return {
+                    id: subChild.id,
+                    label: warnIfMissing(subChild, 'title', 'Missing Title 😭'),
+                    href: warnIfMissing(subChild, 'href', 'Missing Href 😭'),
+                    external: subChild.isExternalLink,
+                    'data-test-subj': getTestSubj(subChild),
+                  };
+                }) ?? [];
+
+            if (child.href) {
+              warnOnce(
+                `Secondary menu item node "${child.id}" has a href "${child.href}", but it should not. We're using it as a section title that doesn't have a link.`
+              );
+            }
+
+            return {
+              id: child.id,
+              label: child.title ?? null, // Use null for no label
+              items: secondaryItems,
+            };
+          })
+        );
+      }
+    }
+
+    warnUnsupportedNavNodeOptions(navNode);
+
+    return {
+      id: navNode.id,
+      label: warnIfMissing(navNode, 'title', 'Missing Title 😭'),
+      iconType: warnIfMissing(navNode, 'icon', AppDeepLinkIdToIcon[navNode.id] || 'faceSad'),
+      // TODO: For now, href shouldn't be required when rendering to open a panel.
+      href: secondarySections?.length ? '' : warnIfMissing(navNode, 'href', 'missing-href-😭'),
+      sections: secondarySections,
+      'data-test-subj': getTestSubj(navNode),
+    };
+  };
+
+  const primaryItems = filterEmpty(primaryNodes.flatMap(toMenuItem));
+
+  const footerItems = filterEmpty(footerNodes.flatMap(toMenuItem));
+
+  if (footerItems.length > 5) {
+    warnOnce(
+      `Footer items should not exceed 5. Found ${
+        footerItems.length
+      }. Only the first 5 will be displayed. Dropped items: ${footerItems
+        .slice(5)
+        .map((item) => item.id)
+        .join(', ')}`
+    );
+  }
+
+  return { logoItem, navItems: { primaryItems, footerItems } };
+};
+
+// =====================
+// Utilities & Helpers
+// =====================
```

Comment:
Nit: separate file for these? I believe it could help to keep this file smaller, even if only slightly